### ISSUE TITLE
Dynamic Scripting With Watcher - JSON Key-Value Inversed

### ISSUE THREAD

**Issue by hasancansaral** [elastic/elasticsearch#18626]
**Elasticsearch version**:
1.5

**JVM version**:
1.8.0_91

**OS version**:
Ubuntu 14.04 LTS Desktop

**Description of the problem including expected versus actual behavior**:

**Steps to reproduce**:
 1. Create an index `PUT some_type`
 2. Create a watcher:

     ```
     {
       "trigger": {
         "schedule": {
           "interval": "5s"
         }
       },
       "input" : {
         "search" : {
           "request" : {
             "indices" : [ "<some_index>" ],
             "types" : [ "<some_type>" ],
             "body" : {
               "query" : {
                 "range": {
                   "_timestamp": {
                     "gte": "now-5s"
                   }
                 }
               }
             }
           }
         }
       },
       "transform" : {
         "script" : "return [ json: groovy.json.JsonOutput.toJson(ctx.payload.hits.hits)]"
       },
       "actions" : {
         "some_webhook" : {
           "webhook" : {
             "method" : "POST",
             "host" : "<some_ip>",
             "port" : <some_port>,
             "path": "<some_endpoint>",
             "body" : "{{ctx.payload.json}}"
           }
         }
       }
     }
     ```

 3. When nothing matches the search, the output is:

    `{ '0': '' }`

 4. When a document matches the search, the output is:

`{ '{"_index":"<some_index>","_type":"<some_type>","_source":{"<some_key>":"<some_value>", ...},"_id":"<doc_id>","_score":1.0}': '' }`

Where `ctx.payload.json` should already be a valid JSON without any need for a key, since `ctx.payload.hits.hits` should be a valid JSON array.

Basically, the data is being posted as the key of the JSON, if a document matches the search, and again '0' as the key of the JSON when no documents match the search. I would appreciate any help, thanks.

**Comment by hasancansaral** [comment]
My bad. I was forgetting:

```
"headers" {
    "Content-type": "application/json"
}
```

so nothing was able to parse (even print) it correctly.
